### ISSUE TITLE
Move 'date' from type IField.type to Attributes.type

### ISSUE THREAD

**Issue by sebastianfiss-consid** [arabdevelop/svelte-formly#65]
The 'date' in IField.type is not handled correctly. There is no component for Date, so nothing will be rendered if I specify that. Furthermore, I can't specify it as a type for the input component.

Example:
```
	const fields: IField[] = [
		{
			type: 'date', // wont work
			name: 'dob',
			attributes: {
				// type: 'date', // cant specify that
				id: 'dob', // required
				classes: ['form-control'],
				placeholder: 'DOB',
				autocomplete: 'off'
			},
		}
	];

...

<Formly fields={fields} {form_name} on:submit={onSubmit} />
```

Fix in [types.ts](https://github.com/dyaliCode/svelte-formly/blob/0307b83c3db928b293820b2d8e1f603a361a2657/src/lib/utils/types.ts#L28)
```
export interface IField {
    type: 'input' | 'textarea' | 'select' | 'checkbox' | 'radio' | 'file' | 'autocomplete'; // remove 'date'
}

export interface Attributes {
    id: string;
    type?: 'text' | 'password' | 'email' | 'number' | 'tel' | 'range' | 'date'; // add 'date'
``

**Comment by kamalkech** [comment]
the date field type is an in input category so not should :

```typescript
const fields = [
  {
    type: 'input', 
    name: 'dob',
    attributes: {
      type: 'date',
      id: 'dob',
      classes: ['form-control'],
      placeholder: 'DOB',
      autocomplete: 'off'
    },
  }
];
```

**Comment by sebastianfiss-consid** [comment]
Yes, but that is not reflected in the exported types. I.e., the type definitions need to change. Currently, it is specified as a possible value for IField.type, but not as an Attribute.type.

**Comment by kamalkech** [comment]
the idea is to force developer respect the struct types of fields, type here:
![image](https://github.com/dyaliCode/svelte-formly/assets/1313428/ad373d53-9d84-4188-80c0-c341eabc1ffe)

which mean you don't need to mention type in attributes object, same thing with field "textarea", "select", "file" and "autocomplete", for more details check documentation [here](https://www.formly-js.com/components/autocomplete)

**Comment by sebastianfiss-consid** [comment]
I do understand your intend, I am saying that it does not work :)

See [`ListFieldType `](https://github.com/dyaliCode/svelte-formly/blob/0307b83c3db928b293820b2d8e1f603a361a2657/src/lib/components/Formly.svelte#L30) in your code: There is no support for a `IField.type = 'date'` (There is also no Date.svelte component). You already mentioned in your previous comment that "the date field type is an in input". So the types should reflect both of that, as I described in my initial issue description. 

If I find the time, I'll look into creating the PR, but I did specify the Fix in the description as well.

The alternative would ofc be to have a Date.svelte component.

**Comment by kamalkech** [comment]
@sebastianfiss-consid i got you and i will fixe this issue soon because i will update the package for current version of svelte.
Thank you for your post and i will be happy if you share any comments or proposition to make this package more better.
